File: JinJiWeb/ClientSite/src/app/shared-models/consumable.ts
export enum ConsumableType {
    食材 = 0,
    耗材,
    每日支出
}

export class Consumable {
    id: number;
    type = ConsumableType.食材;
    year: number;
    month: number;
    day: number;
    name: string;
    description: string;
    cost: number;
    date: string;

    constructor(date: Date) {
        this.date = `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
    }
}

File: JinJiWeb/Models/Consumable.cs
using JinJiWeb.Models.Enums;
using System.ComponentModel.DataAnnotations;

namespace JinJiWeb.Models {
    public class Consumable {
        [Required]
        public int Id { get; set; }

        [Required]
        public int Year { get; set; }

        [Required]
        public int Month { get; set; }

        [Required]
        public int Day { get; set; }

        [Required]
        public ConsumableType Type { get; set; }

        [Required]
        public string Name { get; set; }

        public string Description { get; set; }

        [Required]
        public double Cost { get; set; }
    }
}